src/pages/playground/Playground.tsx

```diff
@@ -138,27 +139,30 @@ export type StateProps = {
 
 const keyMap = { goGreen: 'h u l k' };
 
-function handleHash(hash: string, props: PlaygroundProps) {
+export function handleHash(hash: string, props: PlaygroundProps) {
   const qs = parseQuery(hash);
 
-  const programLz = qs.lz ?? qs.prgrm;
-  const program = programLz && decompressFromEncodedURIComponent(programLz);
-  if (program) {
-    props.handleEditorValueChange(program);
-  }
-
   const chapter = stringParamToInt(qs.chap) || undefined;
-  const variant: Variant =
-    sourceLanguages.find(
-      language => language.chapter === chapter && language.variant === qs.variant
-    )?.variant ?? 'default';
-  if (chapter) {
-    props.handleChapterSelect(chapter, variant);
-  }
+  if (chapter && isNativeJSChapter(chapter)) {
+    showNativeJSWarningOnUrlLoad();
+  } else {
+    const programLz = qs.lz ?? qs.prgrm;
+    const program = programLz && decompressFromEncodedURIComponent(programLz);
+    if (program) {
+      props.handleEditorValueChange(program);
+    }
+    const variant: Variant =
+      sourceLanguages.find(
+        language => language.chapter === chapter && language.variant === qs.variant
+      )?.variant ?? 'default';
+    if (chapter) {
+      props.handleChapterSelect(chapter, variant);
+    }
 
-  const execTime = Math.max(stringParamToInt(qs.exec || '1000') || 1000, 1000);
-  if (execTime) {
-    props.handleChangeExecTime(execTime);
+    const execTime = Math.max(stringParamToInt(qs.exec || '1000') || 1000, 1000);
+    if (execTime) {
+      props.handleChangeExecTime(execTime);
+    }
```

Comment:
I am not quite sure if it would be better UX if we should still load the program string in the default chapter selected when it is intended for `fullJS`? I am 50/50 on this, what do you think?

Reply:
Hmm then it would be too easy to execute shared program, since users can simply switch to `fullJS` later... I am not sure either.

Reply:
I'm voting for the cleanest, simplest solution, which I think is the one you have implemented.